scripts/main.js

```diff
@@ -0,0 +1,191 @@
+    toggleEditMode(row.firstChild); // change back to display mode
+
+}
+
+
+function createNewItem(title, date) {
+    let length = Number(localStorage.getItem("length"));
+    const newItem = {
+        title: title,
+        date: date,
+        status: false,
+        index: length++,
+    }
+    localStorage.setItem("length", length);
+    localStorage.setItem(newItem.index.toString(), JSON.stringify(newItem))
+    renderItem(newItem);
+
+}
+
+function removeItem(row) {
+    const key = row.getAttribute("key")
+    let length = localStorage.getItem("length");
+    let next = row
+    localStorage.removeItem(key);
+    for (let i = Number(key) + 1; i < length; i++) { // undate succesive rows
+        next = next.nextElementSibling;
+        updateItemIndex(i, next);
+    }
+    length--;
+    localStorage.setItem("length", length);
+
+    document.getElementById("todo-list-table").deleteRow(Number(key) + 2);
+}
+
+/**
+ * change between edit and display mode
+ * @param {*} titleNode  - container of elements to toggle
+ */
+function toggleEditMode(titleNode) {
+    const toggleElement1 = titleNode.lastChild.childNodes[0];
+    const toggleElement2 = titleNode.lastChild.childNodes[1];
+    toggleElement1.style.display = toggleElement1.style.display === "none" ? "block" : "none"
+    toggleElement2.style.display = toggleElement2.style.display === "none" ? "block" : "none"
+
+}
+function updateItemIndex(key, itemElement) {
+    const item = JSON.parse(localStorage.getItem(key.toString()));
+    localStorage.removeItem(key.toString())
+    item.index--;
+    localStorage.setItem(item.index.toString(), JSON.stringify(item));
+    itemElement.setAttribute("key", item.index);
+
+}
+
+function updateStatus(row) {
+    const key = row.getAttribute("key");
+    const item = JSON.parse(localStorage.getItem(key));
+    item.status = !item.status; // update the status of the task (done or not)
+    localStorage.setItem(key, JSON.stringify(item));
+    if (row.lastElementChild.textContent) {
+        row.lastElementChild.textContent = "";
+    }
+    else {
+        row.lastElementChild.textContent = checkIcon;
+    }
+}
+function startDrag(event) {
+    row = event.target.parentElement; // mark row we want to move
+}
+function dragOver(event) {
+    const currentRow = event.target.parentElement
+    event.preventDefault();
+    let children = Array.from(currentRow.parentElement.children);
+    if (children.indexOf(currentRow) > children.indexOf(row)) // switch between rows
+        currentRow.after(row);
+    else
+        currentRow.before(row);
+}
+function init() {
+    if (localStorage.length === 0) {
+        localStorage.setItem("length", "0");
+        createNewItem("Hi! Im a new task on your todo list", "03-05-2023")
+        return;
+    }
+    const length = localStorage.getItem("length");
+    for (let key = 0; key < length; key++) {
+        renderItem(JSON.parse(localStorage.getItem(key.toString())));
+    
+    }
+    
+}
```

Comment:
As discussed, split logic; maybe extract this to a function called "loadItemsFromStorage"